### PR TITLE
Probable fix to current DbConnection issues

### PR DESCRIPTION
**Pull request by JulianusIV**
Updated InitConnection to wait for the resource before checking connection status to avoid infinite loops

(also removed the goto loop, and replaced it with a while)

### CODE REVIEW

File: src/CookieTracker.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@ public sealed class CookieTracker : IAsyncDisposable
         private readonly Dictionary<Ulid, CachedCookie> _cachedCookies = new();
         private readonly NpgsqlConnection _databaseConnection;
         private readonly ILogger<CookieTracker> _logger;
+        private readonly SemaphoreSlim _dbConnectionSemaphore = new(1, 1);
```
Comment (OoLunar) [inline src/CookieTracker.cs:23]: Why are we creating a second semaphore slim? If we don't use the original one, then operations will still attempt to execute when the connection could still be broken

File: src/HttpLogger.cs [hunk 2]
```diff
@@ -22,6 +22,9 @@ public sealed class HttpLogger : IServerCompanion
         internal static readonly Action<Microsoft.Extensions.Logging.ILogger, int, string, Exception?> RegisterSlashCommandsFailed = LoggerMessage.Define<int, string>(LogLevel.Error, new EventId(2, "SlashCommands failed to register"), "Failed to register slash commands: {HttpStatusCode} {ReasonPhrase}");
         internal static readonly Action<Microsoft.Extensions.Logging.ILogger, int, Exception?> CookieCreated = LoggerMessage.Define<int>(LogLevel.Debug, new EventId(3, "Cookie Database Operation"), "Created {Count:N0} cookies!");
         internal static readonly Action<Microsoft.Extensions.Logging.ILogger, int, Exception?> CookieUpdated = LoggerMessage.Define<int>(LogLevel.Debug, new EventId(3, "Cookie Database Operation"), "Updated {Count:N0} cookies!");
+        internal static readonly Action<Microsoft.Extensions.Logging.ILogger, Exception?> BakingError = LoggerMessage.Define(LogLevel.Error, new EventId(3, "Cookie Database Operation"), "There was an error baking cookies!");
+        internal static readonly Action<Microsoft.Extensions.Logging.ILogger, Exception?> DbConnection = LoggerMessage.Define(LogLevel.Information, new EventId(4, "Database"), "Connected to database.");
```
Comment (OoLunar) [inline src/HttpLogger.cs:26]: Should be debug

**Review by OoLunar — APPROVED** [review]
Generally LGTM, please resolve or provide insight with the reviews before I'll merge. Thank you again for your contributions!

File: src/CookieTracker.cs (hunk 1 above)
Comment (JulianusIV) [reply src/CookieTracker.cs:23]: see [this conversation](https://discord.com/channels/1070516376046944286/1103119299188641802/1104774964059570247) on Discord

Comment (DubyaDude) [reply src/CookieTracker.cs:23]: Here's a snippet of that convo
Discord_NrGwgurShv

Comment (OoLunar) [reply src/CookieTracker.cs:23]: Ah, I see. I personally think the ideal solution would be to have a method that's called before running any operations, double checking if the database connection is open before returning. Regardless, I can do that another day as my priorities lay within the command handler right now.

**Pull request merged**